# 코카콜라 주식 가격


### Data

주어진 데이터는 다음과 같습니다.

##### coca-cola.csv: 연도별 코카콜라의 주당 가격을 담은 파일

* Year: 연도
* Average Stock Price: 평균 주당 가격(USD/1주)

##### USDKRW_exchange_rate.csv: 연도별 USD -> KRW 환율을 담은 파일

* Year: 연도
* Average USD/KRW: 평균 환율

### 초기 코드 선택

Python, R 중 본인의 선호 언어에 따라 초기 코드를 선택하세요. Python을 선호한다면 Python 초기 코드를, R을 선호한다면 R 초기 코드만 남기면 됩니다.

In [1]:
# 초기코드 - python
import pandas as pd

# 데이터 로드
coca = pd.read_csv('./data/coca-cola.csv', index_col='Year')
exchange_rate = pd.read_csv('./data/USDKRW_exchange_rate.csv', index_col='Year')

coca.head()

Average Stock Price
Year                     
2019              49.2657
2018              44.2552
2017              41.7697
2016              39.6604
2015              36.3935

In [5]:
coca.describe()

Average Stock Price
count            57.000000
mean             11.133204
std              13.678187
min               0.060400
25%               0.267200
50%               3.969300
75%              18.205700
max              49.265700

In [3]:
exchange_rate.head()

Average USD/KRW
Year                 
2019      1159.287025
2018      1100.726341
2017      1130.823450
2016      1160.496416
2015      1132.720611

In [6]:
exchange_rate.describe()

Average USD/KRW
count        60.000000
mean        756.706861
std         364.659229
min          63.125000
25%         464.118125
50%         788.828702
75%        1103.505736
max        1404.026103

In [ ]:
# 초기코드 - R

# 데이터 로드
coca <- read.csv('./data/coca-cola.csv')
exchange_rate <- read.csv('./data/USDKRW_exchange_rate.csv')

head(coca)

# 과제

주어진 데이터를 이용해 1960년부터 2019년까지, 연도별 코카콜라사의 평균 주당 가격을 원화로 나타내려 합니다. 마지막에 보여야 할 결과는 다음과 같습니다.

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Average Stock Price(KRW)</th>
    </tr>
    <tr>
      <th>Year</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2019</th>
      <td>57113.09</td>
    </tr>
    <tr>
      <th>2018</th>
      <td>48712.86</td>
    </tr>
    <tr>
      <th>2017</th>
      <td>47234.16</td>
    </tr>
    <tr>
      <th>2016</th>
      <td>46025.75</td>
    </tr>
    <tr>
      <th>2015</th>
      <td>41223.67</td>
    </tr>
  </tbody>
</table>

### 할 일

##### 1. 데이터 합치기

위에서 로드한 데이터를 연도(year)를 기준으로 합쳐주세요. 합친 후에는 다음과 같이 나와야 합니다.

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Average USD/KRW</th>
      <th>Average Stock Price</th>
    </tr>
    <tr>
      <th>Year</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2019</th>
      <td>1159.287025</td>
      <td>49.2657</td>
    </tr>
    <tr>
      <th>2018</th>
      <td>1100.726341</td>
      <td>44.2552</td>
    </tr>
    <tr>
      <th>2017</th>
      <td>1130.823450</td>
      <td>41.7697</td>
    </tr>
    <tr>
      <th>...</th>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <th>1962</th>
      <td>130.000000</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1961</th>
      <td>124.791667</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1960</th>
      <td>63.125000</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>

In [7]:
# 코드를 작성해주세요.
df = exchange_rate
df.head()

Average USD/KRW
Year                 
2019      1159.287025
2018      1100.726341
2017      1130.823450
2016      1160.496416
2015      1132.720611

In [11]:
df2 = df.merge(coca, how='left', on='Year')
df2.tail()

Average USD/KRW  Average Stock Price
Year                                      
1964       213.846667               0.0804
1963       130.000000               0.0604
1962       130.000000                  NaN
1961       124.791667                  NaN
1960        63.125000                  NaN

##### 2. 결측치 처리

coca.csv 에는 1960년도부터 1962년까지의 데이터가 없습니다. 따라서 이 결측치를 적절합 값으로 채워야 합니다. 1960년도 ~ 1962년 사이에 `Average Stock Price` 컬럼의 결측치를 `Average Stock Price` 컬럼의 최솟값으로 채워주세요.

결측치를 채운 후에는 다음과 같이 나와야 합니다.

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Average USD/KRW</th>
      <th>Average Stock Price</th>
    </tr>
    <tr>
      <th>Year</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2019</th>
      <td>1159.287025</td>
      <td>49.2657</td>
    </tr>
    <tr>
      <th>2018</th>
      <td>1100.726341</td>
      <td>44.2552</td>
    </tr>
    <tr>
      <th>2017</th>
      <td>1130.823450</td>
      <td>41.7697</td>
    </tr>
    <tr>
      <th>...</th>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <th>1962</th>
      <td>130.000000</td>
      <td>0.0604</td>
    </tr>
    <tr>
      <th>1961</th>
      <td>124.791667</td>
      <td>0.0604</td>
    </tr>
    <tr>
      <th>1960</th>
      <td>63.125000</td>
      <td>0.0604</td>
    </tr>
  </tbody>
</table>

In [12]:
coca.min()

Average Stock Price    0.0604
dtype: float64

In [14]:
df3 = df2.fillna(coca.min())
df3

Average USD/KRW  Average Stock Price
Year                                      
2019      1159.287025              49.2657
2018      1100.726341              44.2552
2017      1130.823450              41.7697
2016      1160.496416              39.6604
2015      1132.720611              36.3935
2014      1053.047067              34.9648
2013      1095.113793              32.9671
2012      1127.089744              29.9604
2011      1108.155264              26.1976
2010      1156.650834              21.5905
2009      1276.004245              18.0445
2008      1101.955893              19.2347
2007       929.096176              18.7761
2006       955.011753              14.8972
2005      1023.998178              14.1256
2004      1144.722856              14.9700
2003      1191.029383              13.9335
2002      1245.977503              15.4394
2001      1289.767389              14.8713
2000      1130.938112              16.7047
1999      1188.630929              18.4150
1998      1404.026103              21.4517
1997       954.484168              18.2057
1996       794.334304              13.3057
1995       764.285188               9.0638
1994       805.127306               6.2812
1993       804.129863               5.8364
1992       783.323101               5.6144
1991       737.418014               3.9693
1990       711.603948               2.7406
1989       674.300605               1.9530
1988       734.612342               1.2484
1987       826.086314               1.3408
1986       884.442725               1.0666
1985       872.320403               0.6750
1984       808.063879               0.5326
1983       776.392281               0.4521
1982       732.023118               0.3139
1981       697.295734               0.2675
1980       628.683488               0.2374
1979       484.000000               0.2625
1978       484.000000               0.2672
1977       484.000000               0.2373
1976       484.000000               0.2523
1975       484.000000               0.2306
1974       404.472500               0.2527
1973       398.321667               0.3898
1972       392.894167               0.3666
1971       347.147500               0.2735
1970       310.555833               0.2043
1969       288.160833               0.1934
1968       276.645000               0.1829
1967       270.516667               0.1471
1966       271.337500               0.1015
1965       266.400833               0.0971
1964       213.846667               0.0804
1963       130.000000               0.0604
1962       130.000000               0.0604
1961       124.791667               0.0604
1960        63.125000               0.0604

In [ ]:
# 코드를 작성해주세요

##### 3. 연도별 코카콜라 사의 주당 가격을 원화로 구하기

위에서 구한 데이터를 이용해 코카콜라 사의 평균 주당 가격을 원화로 구해주세요. 단, 주당 가격은 소수점 둘째 자리까지 반올림해 보여주셔야 합니다.

In [17]:
# 코드를 작성해주세요
df3['Average Stock Price(KRW)'] = round(df3['Average USD/KRW'] * df3['Average Stock Price'], 2)
df3

Average USD/KRW  Average Stock Price  Average Stock Price(KRW)
Year                                                                
2019      1159.287025              49.2657                  57113.09
2018      1100.726341              44.2552                  48712.86
2017      1130.823450              41.7697                  47234.16
2016      1160.496416              39.6604                  46025.75
2015      1132.720611              36.3935                  41223.67
2014      1053.047067              34.9648                  36819.58
2013      1095.113793              32.9671                  36102.73
2012      1127.089744              29.9604                  33768.06
2011      1108.155264              26.1976                  29031.01
2010      1156.650834              21.5905                  24972.67
2009      1276.004245              18.0445                  23024.86
2008      1101.955893              19.2347                  21195.79
2007       929.096176              18.7761                  17444.80
2006       955.011753              14.8972                  14227.00
2005      1023.998178              14.1256                  14464.59
2004      1144.722856              14.9700                  17136.50
2003      1191.029383              13.9335                  16595.21
2002      1245.977503              15.4394                  19237.15
2001      1289.767389              14.8713                  19180.52
2000      1130.938112              16.7047                  18891.98
1999      1188.630929              18.4150                  21888.64
1998      1404.026103              21.4517                  30118.75
1997       954.484168              18.2057                  17377.05
1996       794.334304              13.3057                  10569.17
1995       764.285188               9.0638                   6927.33
1994       805.127306               6.2812                   5057.17
1993       804.129863               5.8364                   4693.22
1992       783.323101               5.6144                   4397.89
1991       737.418014               3.9693                   2927.03
1990       711.603948               2.7406                   1950.22
1989       674.300605               1.9530                   1316.91
1988       734.612342               1.2484                    917.09
1987       826.086314               1.3408                   1107.62
1986       884.442725               1.0666                    943.35
1985       872.320403               0.6750                    588.82
1984       808.063879               0.5326                    430.37
1983       776.392281               0.4521                    351.01
1982       732.023118               0.3139                    229.78
1981       697.295734               0.2675                    186.53
1980       628.683488               0.2374                    149.25
1979       484.000000               0.2625                    127.05
1978       484.000000               0.2672                    129.32
1977       484.000000               0.2373                    114.85
1976       484.000000               0.2523                    122.11
1975       484.000000               0.2306                    111.61
1974       404.472500               0.2527                    102.21
1973       398.321667               0.3898                    155.27
1972       392.894167               0.3666                    144.04
1971       347.147500               0.2735                     94.94
1970       310.555833               0.2043                     63.45
1969       288.160833               0.1934                     55.73
1968       276.645000               0.1829                     50.60
1967       270.516667               0.1471                     39.79
1966       271.337500               0.1015                     27.54
1965       266.400833               0.0971                     25.87
1964       213.846667               0.0804                     17.19
1963

In [26]:
# df4 = df3
df4 = df3.drop(['Average USD/KRW', 'Average Stock Price'], axis=1)
df4

Average Stock Price(KRW)
Year                          
2019                  57113.09
2018                  48712.86
2017                  47234.16
2016                  46025.75
2015                  41223.67
2014                  36819.58
2013                  36102.73
2012                  33768.06
2011                  29031.01
2010                  24972.67
2009                  23024.86
2008                  21195.79
2007                  17444.80
2006                  14227.00
2005                  14464.59
2004                  17136.50
2003                  16595.21
2002                  19237.15
2001                  19180.52
2000                  18891.98
1999                  21888.64
1998                  30118.75
1997                  17377.05
1996                  10569.17
1995                   6927.33
1994                   5057.17
1993                   4693.22
1992                   4397.89
1991                   2927.03
1990                   1950.22
1989                   1316.91
1988                    917.09
1987                   1107.62
1986                    943.35
1985                    588.82
1984                    430.37
1983                    351.01
1982                    229.78
1981                    186.53
1980                    149.25
1979                    127.05
1978                    129.32
1977                    114.85
1976                    122.11
1975                    111.61
1974                    102.21
1973                    155.27
1972                    144.04
1971                     94.94
1970                     63.45
1969                     55.73
1968                     50.60
1967                     39.79
1966                     27.54
1965                     25.87
1964                     17.19
1963                      7.85
1962                      7.85
1961                      7.54
1960                      3.81

##### 4. 결과 저장

채점을 하기 위해, 위에서 구한 `Year`, `Average Stock Price(KRW)`를 현재 파일과 같은 디렉토리(.ipynb 파일이 있는 디렉토리)에 `cocacola_stock_krw.csv`라는 이름으로 저장해주세요. 이때 `cocacola_stock_krw.csv`는 년도(Year)를 기준으로 내림차순 정렬해주셔야 합니다. 저장할 파일 포맷은 하단의 표와 data 디렉토리의 `cocacola_stock_krw(sample).csv` 파일을 참고하세요.

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Average Stock Price(KRW)</th>
    </tr>
    <tr>
      <th>Year</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2019</th>
      <td>57113.09</td>
    </tr>
    <tr>
      <th>2018</th>
      <td>48712.86</td>
    </tr>
    <tr>
      <th>2017</th>
      <td>47234.16</td>
    </tr>
    <tr>
      <th>2016</th>
      <td>46025.75</td>
    </tr>
    <tr>
      <th>2015</th>
      <td>41223.67</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
    </tr>
  </tbody>
</table>

<br/>
<center><저장할 파일 포맷 예시></center>


*주의* 

1. csv 파일의 컬럼은 반드시 Year - Average Stock Price(KRW) 순이어야 합니다.
2. **index 순서(Year 2019 ~ 1960)는 꼭 내림차 순이어야 합니다.**

In [ ]:
# 코드를 작성해주세요

In [23]:
# csv 파일 저장 예시 - python
df4.to_csv('cocacola_stock_krw.csv')

In [ ]:
# csv 파일 저장 예시 - R
write.csv(coca, file='cocacola_stock_krw.csv')